src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.kt

```diff
@@ -306,6 +306,28 @@ class AttendanceService(
       .collect(Collectors.toSet())
   }
 
+  fun getAbsencesForReason2(
+    prisonId: String?,
+    absentReason: AbsentReason?,
+    fromDate: LocalDate,
+    toDate: LocalDate?,
+    period: TimePeriod?
+  ): List<AbsenceDto> {
+
+    val periods = period?.let { setOf(it) } ?: setOf(TimePeriod.PM, TimePeriod.AM)
+    val endDate = toDate ?: fromDate
+
+    val attendances = attendanceRepository
+      .findByPrisonIdAndEventDateBetweenAndPeriodInAndAbsentReason(prisonId, fromDate, endDate, periods, absentReason)
+    val attendanceMap = attendances.associateBy({ it.eventId }, { it })
+
+    val offenderDetails = if (attendanceMap.isNotEmpty()) {
```

Comment:
previous version (above) filtered by attendances anyway, so new version just gets the attendances that we know about rather than getting all attendances for all reasons and filtering